### ISSUE TITLE
Spike: understanding student & mentor surveys [8,5,1.6]

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2310]
It's unclear whether setting up student and mentor surveys on TG is a developer task or an admin task, and what the level of integration is between Surveymonkey and the platform. The point of this ticket is to wrangle expectations from the program side with platform capabilities and try to settle on a plan that may or may not require development work to make happen.

**Comment by hellafitz** [comment]
per Devin, pre-survey should be posted 1/14 at latest. On 4/1, it should switch from pre-survey to post-survey.

**Comment by hellafitz** [comment]
1. Look into it (If you fill out a link and some text, where does that show in the UI? What type of SurveyMonkey integration is there, if any?)
2. Present what's in the codebase to the program team
3. Confirm it'll work (or not)
4. If not, get them to elaborate w/ specifics

**Comment by rsgonzal** [comment]
Hey guys!

Some context here - I did set the mentor pre-survey up for Mackenzie earlier in the fall. Checked in with Alli and she mentioned that if it was copied from last year's survey, it should have all of the correct integrations already configured. From what I remember, it looked like it was but she was going to check with the team to see if they had the info they needed.

As far as UI, here is where it shows up for mentors

![]()

![]()

**Comment by stenington** [comment]
Cool, I think it’s worth demystifying this further so we don’t have to rely on copying past surveys and hoping it’s right. I can take a look at the code, or @Allicolyer if you want to you can.

**Comment by stenington** [comment]
Okay, so I see [this method](https://github.com/Iridescent-CM/technovation-app/blob/qa/app/technovation/season_toggles/survey_links.rb#L62) which looks to me like it would be doing the data integration. It looks to me like the following would be true, but I **haven't confirmed it yet**:

If you provide a survey collector url **with query parameters**, the following data could be automatically populated by the platform:
* email as `[email_value]`
* country as `[country_value]`
* state or province as `[state_value]`
* user's full name as `[name_value]`
* city as `[city_value]`
* age as `[age_value]`
* school or company name as `[school_value]` or as `[company_value]`
* list of team names as `[team_value]`

If you don't include those as query parameters, no data will be sent across. I believe the way you get those query parameters added is to set up **custom variables** on the survey, where the variable name has to match the X in `[X_value]` **exactly**, so `email`, `country`, etc.

The current mentor survey link configured on the platform **does not** have any query params, so the platform probably isn't sending across any additional data.

**Comment by stenington** [comment]
I think we have a good start here. One useful outcome of this ticket could be a slightly cleaner piece of documentation documenting...

* what happens when you set a survey link
* what happens for the user when they take a survey
* what happens for a user when they dismiss a survey modal
* how do you show a second survey (e.g. a post-survey vs. a pre-survey)
* what information can the platform send to Surveymonkey for a user, and
* what does an admin have to do to make that happen?

**Comment by stenington** [comment]
Documentation is underway, we might not be recording survey completion correctly for the mentor pre-survey, but it's not a big deal. It's also unclear if the platform supports two surveys in one season right now without developer intervention. Both of these things will be outlined in the docs.

**Comment by stenington** [comment]
Looks like we might also not be set up to show certain users the pop-up window with the link. I'm not sure how much that affects response rates, but something to keep an eye on. We may want to advertise the survey some other way to encourage responses.

**Comment by stenington** [comment]
I'm building out documentation [here](https://docs.google.com/document/d/1hVaugqKipnD8pYmCcMl5KWCHrrqBBSr-86MQZTdm5P8/edit#)

**Comment by rsgonzal** [comment]
So from my conversation with assessment:
- less questions on the actual survey sounds great
- they think all of the data above would be useful to collect automatically
- most important data is team name because it helps connect with scores and students/mentors/RAs
- collecting a user ID will also be helpful for connecting pre and post surveys but less important than ^
- some pieces of info are important to collect on the actual survey, seemed like just country, they're ok with just having that information twice
- they would like to have a higher completion rate, but don't want the survey to be required - e ways to make it more visible, also talked about editing text to be more encouraging to tak

Questions:
-  if someone starts the survey but doesn't complete it, will this data still be collected (they can see started surveys and get some info, wondering if this data will stick too)
- do we know how many times the pop up shows, would like it to be shown continuously until they take it, and then potentially take down popup in feb

**Comment by stenington** [comment]
Collecting some relevant links here.

I opened #2329 and #2330 to track the specific tasks needed for reconfiguring the mentor pre-survey and configuring the student pre-survey. I think having written records of how we're doing this will help us settle on a process and help highlight additional work we may want to do to improve things.

For historical context, here are two past tickets where we've wrestled with this: #2151 and #1846
